backends/bmv2/common/control.cpp

```diff
@@ -100,6 +107,42 @@ void ControlConverter::convertTableEntries(const IR::P4Table *table,
                 } else {
                     ::error(ErrorType::ERR_UNSUPPORTED, "range key expression", k);
                 }
+            } else if (matchType == "optional") {
+                // Table key fields with match_kind optional with
+                // "const entries" in the P4 source code will be
+                // represented using the same "key" and "mask" keys in
+                // the BMv2 JSON file as table key fields with
+                // match_kind ternary.  We will perform here the more
+                // strict checking of the allowed masks that optional
+                // imposes.
+                if (k->is<IR::Mask>()) {
```

Comment:
do you think we should even allow mask expressions in the P4? We already have `_` for the don't care case. We would no longer need this code then.

Reply:
That makes a lot of sense.  Good review comment!  I will change the code not to use masks for const entries on optional fields at all, but only exactly value or `_`/`default`

Reply:
@antoninbas If you can stand to take another look at this PR, I think I have addressed your two comments.